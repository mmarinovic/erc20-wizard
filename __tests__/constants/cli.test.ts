import {
  cliPromptOptions,
  cliMessages,
  cliOptions,
} from '../../src/constants/cli';

describe('cli', () => {
  it('cliPromptOptions matches a snapshot', () => {
    expect(cliPromptOptions).toMatchSnapshot();
  });

  it('cliMessages matches a snapshot', () => {
    expect(cliMessages).toMatchSnapshot();
  });

  it('cliOptions matches a snapshot', () => {
    expect(cliOptions).toMatchSnapshot();
  });
});
