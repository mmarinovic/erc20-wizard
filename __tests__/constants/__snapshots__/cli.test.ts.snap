// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli cliMessages matches a snapshot 1`] = `
Object {
  "configDisplay": [Function],
  "configureSuccess": "üöÄ Sucessfully configured",
  "deploying": "üî• Deploying üî•",
  "deploymentError": [Function],
  "deploymentSuccess": [Function],
  "help": "
  --help                            Print help
  --config=infuraProjectId          Configure factory to use 'infuraProjectId' when connecting to infura API. Get it at https://infura.io/.
  --main                            Start contract creation for Ethereum main network
  --rinkeby                         Start contract creation for Ethereum rinkeby testnet
  --ropsten                         Start contract creation for Ethereum ropsten testnet
  ",
  "invalidProjectId": "[31m‚ùå Invalid Infura project id[39m",
  "notConfigured": "[31m‚ùå CLI is not configured yet. Use --config to setup your infura project id. Get your project id at https://infura.io/[39m",
  "welcome": "üè≠  Welcome to Ethereum ERC20 token factory! üè≠",
}
`;

exports[`cli cliOptions matches a snapshot 1`] = `
Object {
  "boolean": Array [
    "help",
  ],
  "string": Array [
    "config",
    "network",
  ],
}
`;

exports[`cli cliPromptOptions matches a snapshot 1`] = `
Array [
  Object {
    "description": "[35mPlease enter your private key. This will be used to sign contract transaction.[39m",
    "hidden": true,
    "name": "privateKey",
    "required": true,
  },
  Object {
    "description": "Token name",
    "name": "tokenName",
    "pattern": /\\^\\[a-zA-Z \\]\\+\\$/,
    "required": true,
  },
  Object {
    "description": "Token symbol",
    "name": "tokenSymbol",
    "pattern": /\\^\\[a-zA-Z\\]\\+\\$/,
    "required": true,
  },
  Object {
    "description": "Token decimal places",
    "name": "tokenDecimals",
    "pattern": /\\^\\[0-9\\]\\+\\$/,
    "required": true,
  },
  Object {
    "description": "Token total supply",
    "name": "tokenTotalSupply",
    "pattern": /\\^\\[0-9\\]\\+\\$/,
    "required": true,
  },
]
`;
